app.js

```diff
@@ -0,0 +1,15 @@
+const express = require('express');
```

Comment:
It would be better to put the server startup logic into an async function. This allows you to await connectDB before the server starts, which will only allow handling requests once the database connection is successful.

Here is a way to hanlde that:

`async function startServer() {
  try {
    await connectDB();
    console.log('Database connected successfully');

    app.use('/notes', noteRoutes);

    const PORT = 3000;
    app.listen(PORT, () => {
      console.log(`Server is running on port ${PORT}`);
    });
  } catch (error) {
    console.error('Database connection failed', error);
    process.exit(1);
  }
}

startServer();
`

Reply:
Done